[Docs] New Drivers and ORMs Section (#10878)
yugabyte/yugabyte-db: docs/content/latest/drivers-orms/nodejs/sequelize.md

[Sequelize ORM](https://sequelize.org/v6/)  is a popular ORM provider for Node.js applications which is widely used by Node Developers for Database access.

## CRUD Operations with GORM

Learn how to establish a connection to YugabyteDB database and begin simple CRUD operations using the steps on the [Build an application](../../../quick-start/build-apps/nodejs/ysql-sequelize/) page under the Quick start section.

To break down the quick start example and understand how to perform the common CRUD operations required for Go application development, do the following steps.

### Step 1: Creating a Node.js project and installing Sequelize ORM core package

Before proceeding with the next steps, you need to have Node.js installed on your machine. To install it, please refer to this [link](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm#using-a-node-installer-to-install-node-js-and-npm).

> Now, let's create a simple Node.js project and install a `sequelize` orm core package.

Reviewer: ```suggestion
To create a simple Node.js project and install the `sequelize` core package, do the following:
```

1. Create a new directory: 
`mkdir nodejs-quickstart-example && cd nodejs-quickstart-example`

2. Create a package.json file: 
`echo {} > package.json`

3. Install sequelize-yugabytedb package: 
`npm install sequelize`

4. Install node-postgres driver:
 `npm install pg` 

5. Create a empty demo.js file: 
`touch example.js`

### Step 2: Creating a Node.js example using Sequelize ORM

> Add the following code in the `example.js`, this allows you to create an object for Employees to store and retrieve information of employees from the YugabyteDB. Here, first of all we have created the sequelize connection by passing the basic connection information and dialect as `postgres` then we have defined the object of type Employee using `define()` API which allows us to mention what all type of information we want to store for an employee but the actual table is created when we have called the `Employee.sync()` API in `createTableAndInsert()` function which also inserts the data of three employees into the table employees using `Employee.create()` API. At last, we can retrieve the information of all employees using `Employee.findAll()`.

Reviewer: ```suggestion
Add the following code in the `example.js` file. This code creates an Employees object to store and retrieve employee information. First it creates a connection by passing the basic connection parameters. Next, it defines the Employee model using the `define()` API, which specifies the type of information to store for an employee. The actual table is created by calling the `Employee.sync()` API in the `createTableAndInsert()` function. This also inserts the data for three employees into the table using the `Employee.create()` API. Finally, you can retrieve the information of all employees using `Employee.findAll()`. 
```

    emp_id : {
        type: DataTypes.INTEGER,
    primaryKey: true
    },
    emp_name: {
        type: DataTypes.STRING,
     },
    emp_age: {
        type: DataTypes.INTEGER,
     },
    emp_email:{
        type: DataTypes.STRING,	
      },
   }
)
async function createTableAndInsert() {
   //creating a table “employees”
   await Employee.sync({force: true});
   console.log("Created the employees Table.")

  //Insert 3 rows into the employees table
   
   await Employee.create({emp_id: 1, emp_name: 'sam', emp_age: 25, emp_email: 'sam@example.com'})
   await Employee.create({emp_id: 2, emp_name: 'bob', emp_age: 27, emp_email: 'bob@example.com'})
   await Employee.create({emp_id: 3, emp_name: 'Jake', emp_age: 29, emp_email: 'jake@example.com'})
   console.log("Inserted the data of three employees into the employees table");
}

async function fetchAllRows() {
    //fetching all the rows
    console.log("Fetching the data of all employees.")
    const employees = await Employee.findAll({raw: true});
   console.log("Employees Details: \n", employees);
    process.exit(0)
}

createTableAndInsert()
.then(function() {
    fetchAllRows()
})

```

> When you run the example.js using `node example.js`, you will get this as output-

Reviewer: ```suggestion
When you run example.js using `node example.js`, you should get output similar to the following:
```

Executing (default): INSERT INTO "employees" ("emp_id","emp_name","emp_age","emp_email","createdAt","updatedAt") VALUES ($1,$2,$3,$4,$5,$6) RETURNING "emp_id","emp_name","emp_age","emp_email","createdAt","updatedAt";
Inserted the data of three employees into the employees table
Fetching the data of all employees.
Executing (default): SELECT "emp_id", "emp_name", "emp_age", "emp_email", "createdAt", "updatedAt" FROM "employees" AS "employees";
Employees Details: 
 [
  {
    emp_id: 1,
    emp_name: 'sam',
    emp_age: 25,
    emp_email: 'sam@example.com',
    createdAt: 2022-03-21T09:25:38.936Z,
    updatedAt: 2022-03-21T09:25:38.936Z
  },
  {
    emp_id: 2,
    emp_name: 'bob',
    emp_age: 27,
    emp_email: 'bob@example.com',
    createdAt: 2022-03-21T09:25:38.952Z,
    updatedAt: 2022-03-21T09:25:38.952Z
  },
  {
    emp_id: 3,
    emp_name: 'Jake',
    emp_age: 29,
    emp_email: 'jake@example.com',
    createdAt: 2022-03-21T09:25:38.955Z,
    updatedAt: 2022-03-21T09:25:38.955Z
  }
]
```

> ## Limtations

Reviewer: ```suggestion
## Limitations
```

> YugabyteDB YSQL is compatible with Sequelize ORM's PostgreSQL dialect. Currently, YugabyteDB doesn't have support for Sequelize ORM's `findorCreate()` API, along with other issues, which may prevent users from successfully implementing Node.js applications. There is [on going work](https://github.com/yugabyte/yugabyte-db/issues/11683) to support the Sequelize ORM core package with YugabyteDB natively. In the interim, We recommend users to make use of [sequelize-yugabytedb](../sequelize-yugabytedb) for building Node.js applications.

Reviewer: ```suggestion
YugabyteDB YSQL is compatible with Sequelize ORM's PostgreSQL dialect. Currently, YugabyteDB doesn't support the Sequelize ORM `findorCreate()` API, and some other features, which may prevent you from successfully implementing Node.js applications. There is [ongoing work](https://github.com/yugabyte/yugabyte-db/issues/11683) to add support for YugabyteDB to the Sequelize ORM core package. In the meantime, use [sequelize-yugabytedb](https://github.com/yugabyte/sequelize-yugabytedb) to build Node.js applications.
```

## Next Steps

- Explore [Scaling Node.js Applications](/latest/explore/linear-scalability) with YugabyteDB.

> - Learn how to [develop Node Applications with YugabyteDB Cloud](/latest/yugabyte-cloud/cloud-quickstart/cloud-build-apps/cloud-ysql-node/).

Reviewer: ```suggestion
- Learn how to [develop Node applications with Yugabyte Cloud](/latest/yugabyte-cloud/cloud-quickstart/cloud-build-apps/cloud-ysql-node/).
```
